File: GeeksforGeeks_Solutions/python/October_02.py
class Solution:
    def rotateDelete(self,  arr):
        var1 = 0
        var2 = 0
        var3 = len(arr)-1

        while(len(arr) != 1):
            x = arr[len(arr) - 1]
            arr.insert (0, x)
            arr.pop (len(arr) - 1)
            y = var3 - var1 -var2
            y = 0 if y<0 else y
            arr.pop(y)
            var1 += 1
            var2 += 1
        return arr[0]

File: GeeksforGeeks_Solutions/C++/October_02.cpp
#include <iostream>
using namespace std;

class Solution
{
public:
    int rotateDelete(vector<int> &arr)
    {
        int n = arr.size();
        int k = 1;
        while (n > 1)
        {
            arr.insert(arr.begin(), arr.back()); // inserting last ele.to front
            arr.pop_back();                      // removing back ele.

            if (k > n)
                arr.erase(arr.begin()); // greater than n,delete 1st ele
            else
                arr.erase(arr.end() - k); // delete n-k+1th ele. from begin

            n = arr.size(); // update n
            k++;            // inc k
        }

        return arr[0];
    }
};